### ISSUE TITLE
Instagram likes do not link to webmention target

### ISSUE THREAD

**Issue by capjamesg** [snarfed/bridgy#1064]
The Instagram likes sent by Brid.gy to my domain do not include links to the article to which they are sent. This means I cannot process them as webmention sources must contain the target link.

Here is the h-entry of one like sent to my site (spaces removed):

```
<article class="h-entry">
  <span class="p-uid">tag:instagram.com,2013:2618243262510956500_41923765797_liked_by_44226013261</span>

  <span class="p-author h-card">
    <data class="p-uid" value="tag:instagram.com,2013:44226013261"></data>
    <a class="p-name u-url" href="https://www.instagram.com/dotabean/">DotaBean</a>
<a class="u-url" href="https://www.instagram.com/44226013261/"></a>
    <span class="p-nickname">dotabean</span>
    <img class="u-photo" src="https://aujtzahimq.cloudimg.io/v7/https://instagram.fgla3-1.fna.fbcdn.net/v/t51.2885-19/s150x150/202390866_122853073295441_2556871516123227135_n.jpg?_nc_ht=instagram.fgla3-1.fna.fbcdn.net&_nc_ohc=kOhU4iD1X1YAX-zvh7m&edm=AAo1ks0BAAAA&ccb=7-4&oh=de60728e3b44c93cc7b0a8567176287c&oe=61214413&_nc_sid=01e9e1" alt="" />
  </span>

  <a class="p-name u-url" href="https://www.instagram.com/p/CRV3mANhn_U/#liked-by-44226013261"></a>
  <div class="">
  </div>

<a class="u-like-of" href="https://www.instagram.com/p/CRV3mANhn_U/"></a>

</article>
```

The full source is here: view-source:https://brid.gy/like/instagram/capjamesg/2618243262510956500_41923765797/44226013261

I think the u-like-of attribute should point to my domain so that I can process the webmention.

**Comment by snarfed** [comment]
Ugh. Thanks for filing! Discussed on chat, log below. Looks like IG may have changed some of their JSON data and switched around which parts have user ids vs usernames, so we're not connecting them quite right any more. Fix may have to wait until the Flask port is done, sadly. Sorry for the delay!

```
discovering original posts for: https://www.instagram.com/p/CRV3mANhn_U/
Original post discovery found original posts {'https://jamesg.blog/2021/07/15/coffee-food-pairings'}, mentions set()
Demoting original post links because user tag:instagram.com,2013:capjamesg doesn't match author tag:instagram.com,2013:41923765797"
starting posse post discovery with syndicated https://www.instagram.com/p/CRV3mANhn_U/
```

**Comment by capjamesg** [comment]
I also noticed Instagram likes to not contain a dt-published attribute. Perhaps this is caused by the same issue you mentioned.

I have decided to overlook the requirement for a source to contain a link to a target on https://brid.gy/like/instagram/* pages for now so that I can backfeed likes onto my blog. I will remove this logic (since it technically goes against the webmention standard) once everything is working.

Thanks for everything (including building brid.gy -- it is incredible!).
